### PR TITLE
Enable review suppression on success

### PR DESCRIPTION
**Pull request by orgads**

### CODE REVIEW

**Comment by orgads** [comment]
This conflicts with #27. After you accept one of them, I'll rebase the other on top of it. I prefer to have #27 before this one, less noise.

File: src/main/resources/org/jenkinsci/plugins/sonargerrit/SonarToGerritPublisher/config.properties [hunk 1]
```diff
@@ -5,6 +5,8 @@ jenkins.plugin.settings.section.filter.name=Filter Settings
 jenkins.plugin.settings.filter.severity=Report issues having severity level higher or equal to:
 jenkins.plugin.settings.filter.new=Report new issues only?
 jenkins.plugin.settings.filter.new.description=Reports new as well as existing issues when unchecked
+jenkins.plugin.settings.filter.failonly=Report only on failure?
+jenkins.plugin.settings.filter.failonly.description=Reports only failures when checked
```
Comment (mkp05) [inline src/main/resources/org/jenkinsci/plugins/sonargerrit/SonarToGerritPublisher/config.properties]: I think you can elaborate/update the description giving more details - what kind of failures do you mean?

Comment (orgads) [inline src/main/resources/org/jenkinsci/plugins/sonargerrit/SonarToGerritPublisher/config.properties]: Any failure. I mean to suppress the report on success.

Comment (mkp05) [inline src/main/resources/org/jenkinsci/plugins/sonargerrit/SonarToGerritPublisher/config.properties]: Usually in Jenkins stuff "failure" (personally for me) means "build **_failure_**" not, like I think in this case Sonar **_issue_**.

Comment (orgads) [inline src/main/resources/org/jenkinsci/plugins/sonargerrit/SonarToGerritPublisher/config.properties]: Done

File: src/main/java/org/jenkinsci/plugins/sonargerrit/SonarToGerritPublisher.java [hunk 2]
```diff
@@ -133,6 +133,10 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull
                 TaskListenerLogger.logMessage(listener, LOGGER, Level.INFO, "jenkins.plugin.issues.to.score", file2issuesToScore.entries().size());
             }
 
+            if (file2issuesToComment.isEmpty() && file2issuesToScore.isEmpty() && reviewConfig.getIssueFilterConfig().getFailOnly()) {
```
Comment (orgads) [inline src/main/java/org/jenkinsci/plugins/sonargerrit/SonarToGerritPublisher.java]: reviewConfig here is null! How is it possible?

Comment (orgads) [reply src/main/java/org/jenkinsci/plugins/sonargerrit/SonarToGerritPublisher.java]: Or maybe one of these file2issues is null...

**Comment by orgads** [comment]
ping? This now works for me.

**Comment by orgads** [comment]
Nope, compilation is broken...

**Comment by orgads** [comment]
Works now. PING!

**Comment by aquarellian** [comment]
Hi, thank you for the contribution!
I think, IssueFilterConfig is not the best place for the setting, ReviewConfig would do better.
Will get back to you on the next matter. SonarToGerritPublisher is also not a best place for the business logic. Will be better to handle it in GerritRevisionWrapper/GerritReviewBuilder, but not quite sure, if skipping sending review will not broke GerritTrigger logic.

**Comment by orgads** [comment]
[**@aquarellian**](https://github.com/aquarellian) commented on [Jan 6, 2018, 11:43 PM GMT+2](https://github.com/jenkinsci/sonar-gerrit-plugin/pull/25#issuecomment-355780164 "2018-01-06T21:43:48Z - Replied by Github Reply Comments"):
> Hi, thank you for the contribution!  
> 
> I think, IssueFilterConfig is not the best place for the setting, ReviewConfig would do better.  

I started with ReviewConfig, but went back to IssueFilterConfig because ReviewConfig is about _templates_, which have nothing to do with this filtering. The `Review Settings` pane makes much more sense for this setting.

**Comment by aquarellian** [comment]
Hi there,
though I think that you are right about necessity of separating of templates, from my point of view the setting you propose is still fits ReviewConfig rather than Filter: it have nothing to do with filter, either. In addition, filter is used for other purposes where this setting will make no sense. Thus, I think, the best way to decompose the Review settings is to add 'Review suppression' to ReviewConfig and extract a new ReportConfig setting for all the templates.

**Pull request closed without merging**